test/Base_Tests/src/Data/Dictionary_Spec.enso

```diff
@@ -269,6 +270,11 @@ add_specs suite_builder =
             m.at 200 . should_equal 3
             m.at Nothing . should_equal 1
 
+        group_builder.specify "should attach a warning when a Float is used as a key" <|
+            m = Dictionary.empty . insert 1.2 3
+            m.at 1.2 . should_equal 3
+            Problems.expect_only_warning Float_Used_As_Dictionary_Key m
```

Comment:
Please also add a test that checks if we have the same behaviour when we construct the dictionary using `Dictionary.from_vector`.

Reply:
Done